packages/lib-classifier/src/plugins/drawingTools/models/marks/Ellipse/Ellipse.js

```diff
@@ -0,0 +1,95 @@
+import { getParentOfType, types } from 'mobx-state-tree'
+import { Ellipse as EllipseComponent } from '@plugins/drawingTools/components/'
+import { EllipseTool } from '@plugins/drawingTools/models/tools'
+
+import Mark from '../Mark'
+
+const BUFFER = 24
+const DELETE_BUTTON_ANGLE = -45
+const MINIMUM_RADIUS = 5
+
+const EllipseModel = types
+  .model('EllipseModel', {
+    x_center: types.maybe(types.number),
+    y_center: types.maybe(types.number),
+    rx: types.optional(types.number, 0),
+    ry: types.optional(types.number, 0),
+    angle: types.optional(types.number, 0)
+  })
```

Comment:
Let's add `toolType: types.literal('ellipse')` here

Reply:
Out of interest, why `toolType` and not `type`?

Reply:
This should be added to all marks in a separate PR, since it changes how snapshots are handled for all marks.

Reply:
To clearly distinguish it from the task type. We've had enough trouble keeping straight that it was only the task models that had types defined on them and not the annotation and mark models, so the more specific naming, the better, imo.

Reply:
A separate PR is ok. Can you open an issue to track it?